Oracle.md

```diff
@@ -0,0 +1,190 @@
+        - [Version 0 `external_event_descriptor`](#version-0-external_event_descriptor)
+        - [Version 0 `enum_event_descriptor`](#version-0-enum_event_descriptor)
+        - [Version 0 `range_event_descriptor`](#version-0-range_event_descriptor)
+        - [Version 0 `large_range_event_descriptor`](#version-0-large_range_event_descriptor)
+    - [Oracle events](#oracle-events)
+        - [Version 0 `oracle_event`](#version-0-oracle_event)
+- [Oracle announcements](#oracle-announcements)
+        - [Version 0 `oracle_announcement`](#version-0-oracle_announcement)
+
+
+## Event descriptor
+
+An event descriptor provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Four kinds of event outcomes are defined (note that a single event can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the outcomes can simply be enumerated.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- count: the number of possible outcomes
+- step: the increment
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base 10 exponent by which to multiply the signed number to obtain the actual outcome value.
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+count: 201
+step: 1
+unit: Celcius degrees
+precision: 0
+```
+
+### Large range
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using numerical decomposition.
+
+The event descriptor should include:
+
+- base: the base in which the outcome value is decomposed and which will indicate the possible range of each digit that will be signed.
+- is-signed: a boolean value indicating whether the outcomes can be negative.
+- r-values: an array of R values, one for each of the digit that will be signed by the oracle, with the first value of the array being used for signing the rightmost digit. If the is-signed value is true, an additional R-value must be provided as the first element in this array, which will be used to sign the string "+" in case of an outcome with a positive value or zero value and the string "-" in case of an outcome with a negative one.
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base 10 exponent by which to multiply the number represented by the composition of the digits to obtain the actual outcome value.
```

Comment:
Shouldn't be the exponent in the same encoding than the base under exponentiation ? Otherwise if the oracle you need to spec base encoding to be sure conversions are correct. 

E.g, where XYZ is used by Alice and RST by Bob you might have the following:
```
version_XYZ_base64(base10-exponent) != version_RST_base64(base10-exponent)
```

Reply:
I had already updated the sentence by removing the `10`, but I'm not sure I fully understand your concern, is it ok like that?